Services/Dashboard/Block/classes/class.ilDashboardBlockGUI.php

```diff
@@ -0,0 +1,683 @@
+            $data->getDescription(),
+            $data->getTitle(),
+            $data->getDescription()
+        );
+
+        return $card;
+    }
+
+    protected function getListItemGroups(): array
+    {
+        $data = $this->loadData();
+        $groupedCards = [];
+        foreach ($data as $title => $group) {
+            $items = [];
+            foreach ($group as $datum) {
+                $item = $this->getListItemForDataDTO($datum);
+                if ($item !== null) {
+                    $items[] = $item;
+                }
+            }
+            $groupedCards[] = $this->factory->item()->group((string) $title, $items);
+        }
+
+
+        return $groupedCards;
+    }
+
+    protected function getListItemForDataDTO(ilBlockDTO $data): ?Item
+    {
+        $itemListGui = $this->byType($data->getType());
+        $this->addCustomCommandsToActionMenu($itemListGui, $data->getRefId());
+        $list_item = $itemListGui->getAsListItem(
+            $data->getRefId(),
+            $data->getObjId(),
+            $data->getDescription(),
+            $data->getTitle(),
+            $data->getDescription()
+        );
+
+        return $list_item;
+    }
+
+    protected function isRepositoryObject(): bool
+    {
+        return false;
+    }
+
+    protected function getLegacyContent(): string
+    {
+        $groupedCards = [];
+        foreach ($this->loadData() as $title => $group) {
+            $cards = array_filter(array_map($this->getCardForData(...), $group));
+            if ($cards) {
+                $groupedCards[] = $this->ui->factory()->panel()->sub(
+                    $title,
+                    $this->factory->deck($cards)->withNormalCardsSize()
+                );
+            }
+        }
+
+        if ($groupedCards) {
+            return $this->renderer->render($groupedCards);
+        }
+
+        return $this->getNoItemFoundContent();
+    }
+
+    public function getNoItemFoundContent(): string
+    {
+        return $this->emptyHandling();
+    }
+
+    public function getViewSettings(): ilPDSelectedItemsBlockViewSettings
+    {
+        return $this->viewSettings;
+    }
+
+    public function init(): void
+    {
+        $this->initViewSettings();
+        $this->main_tpl->addJavaScript('Services/Dashboard/Block/js/replaceModalContent.js');
+        $this->viewSettings->parse();
+        $this->requested_item_ref_id = (int) ($this->http->request()->getQueryParams()["item_ref_id"] ?? 0);
+        $this->initData();
+
+        $this->ctrl->setParameter($this, 'view', $this->viewSettings->getCurrentView());
+        if ($this->viewSettings->isTilePresentation()) {
+            $this->setPresentation(self::PRES_MAIN_LEG);
+        } else {
+            $this->setPresentation(self::PRES_MAIN_LIST);
+        }
+    }
+
+    protected function initAndShow(): void
+    {
+        $this->init();
+        if ($this->ctrl->isAsynch()) {
+            echo $this->getHTML();
+            exit;
```

Comment:
Please use `$DIC->http()` to send a response and `exit` (with `$DIC->http()->close()` etc).